File: java/src/main/java/com/epam/eng/pattern/decorator/Coffee.java
package com.epam.eng.pattern.decorator;

public class Coffee {
    private String size;
    private boolean milk;
    private boolean sugar;
    private double cost;

    public Coffee(String size, boolean milk, boolean sugar, double cost) {
        this.size = size;
        this.milk = milk;
        this.sugar = sugar;
        this.cost = cost;
    }

    public String getDescription() {
        String description = size;
        if (milk) {
            description += " with milk";
        }
        if (sugar) {
            description += " with sugar";
        }
        return description;
    }

    public double getCost() {
        return cost;
    }

}

File: ts/decorator/Coffee.ts
class Coffee {
    private size: string;
    private milk: boolean;
    private sugar: boolean;
    private cost: number;

    constructor(size: string, milk: boolean, sugar: boolean, cost: number) {
        this.size = size;
        this.milk = milk;
        this.sugar = sugar;
        this.cost = cost;
    }

    public getDescription(): string {
        let description = this.size;
        if (this.milk) {
            description += " with milk";
        }
        if (this.sugar) {
            description += " with sugar";
        }
        return description;
    }

    public getCost(): number {
        return this.cost;
    }
}

// todo: consider to decorator Coffee with sugar, with milk.

